misc/systemd/ipfs.service

```diff
@@ -0,0 +1,31 @@
+[Unit]
+description=ipfs p2p daemon
+After=network.target
+Requires=network.target
+
+[Service]
+Type=simple
+User=ipfs
+RestartSec=1
+Restart=always
+PermissionsStartOnly=true
+Nice=18
+StateDirectory=/var/lib/ipfs
+Environment=IPFS_PATH=/var/lib/ipfs
+Environment=HOME=/var/lib/ipfs
+LimitNOFILE=8192
+Environment=IPFS_FD_MAX=8192
+EnvironmentFile=-/etc/sysconfig/ipfs
+StandardOutput=journal
+WorkingDirectory=/var/lib/ipfs
+ExecStartPre=-adduser --system --group --home /var/lib/ipfs ipfs
+ExecStartPre=-mkdir /var/lib/ipfs
+ExecStartPre=-/bin/chown ipfs:ipfs /var/lib/ipfs
+ExecStartPre=-/bin/chmod u+rwx /var/lib/ipfs
+ExecStartPre=-chpst -u ipfs /usr/local/bin/ipfs init --profile=badgerds
```

Comment:
There is no way to run some of the `ExecStartPre`s as root (e.g. the adduser/chown/chmod) and others as the user in question (`ipfs`).  Either all are run as root or all are run as the `User` specified (depending on `PermissionsStartOnly`).  Because root is required for making sure the user exists and the directory has the correct owner/permissions, chpst is required to do the `ipfs init` as the correct user.  The fact that it is not installed by default is not our concern; ipfs is also not installed by default and `/usr/local/bin/ipfs` will not be available unless the user installs that.

Reply:
Additionally, if someone tries to use this unit file and it fails, checking the systemd journal will show that they need `chpst` quite plainly.